Changelog entry for v0.8.0 (#2272)
GenericMappingTools/pygmt: doc/changes.md

# Changelog

> ## Release v0.8.0 (2022/12/29)

Reviewer: ```suggestion
## Release v0.8.0 (2022/12/30)
```

* Added an intro tutorial for creating contour maps ([#2126](https://github.com/GenericMappingTools/pygmt/pull/2126))

### New Features

* Add load_earth_free_air_anomaly function for Earth free-air anomaly dataset ([#2238](https://github.com/GenericMappingTools/pygmt/pull/2238))
* Add load_earth_geoid function for Earth Geoid dataset ([#2236](https://github.com/GenericMappingTools/pygmt/pull/2236))
* Add load_earth_magnetic_anomaly function for Earth magnetic anomaly dataset ([#2196](https://github.com/GenericMappingTools/pygmt/pull/2196), [#2239](https://github.com/GenericMappingTools/pygmt/pull/2239))
* Add load_earth_vertical_gravity_gradient function for Earth vertical gravity gradient dataset ([#2240](https://github.com/GenericMappingTools/pygmt/pull/2240))
* load_earth_relief: Add the support of data sources 'gebco' , 'gebcosi', and 'synbath' ([#1818](https://github.com/GenericMappingTools/pygmt/pull/1818), [#2162](https://github.com/GenericMappingTools/pygmt/pull/2162), [#2192](https://github.com/GenericMappingTools/pygmt/pull/2192))
* Wrap ternary ([#1431](https://github.com/GenericMappingTools/pygmt/pull/1431))

### Enhancements

> * Add ternary sample dataset ([#2211](https://github.com/GenericMappingTools/pygmt/pull/2211))

Reviewer 2: ```suggestion
* Set gridline (if available) as the default grid registration for remote datasets ([#2266](https://github.com/GenericMappingTools/pygmt/pull/2266))
* Add ternary sample dataset ([#2211](https://github.com/GenericMappingTools/pygmt/pull/2211))
```

# Changelog

## Release v0.8.0 (2022/12/28)

[![Digital Object Identifier for PyGMT v0.8.0](https://zenodo.org/badge/DOI/10.5281/zenodo.7481934.svg)](https://doi.org/10.5281/zenodo.7481934)

### Highlights

* 🎉 **Eighth minor release of PyGMT** 🎉
* Wrapped ternary

> *

Reviewer 3: ```suggestion
* Add intro tutorial section for creating contour maps
* Easy access for various remote datasets
```
maybe the formulation needs to be polished a litte bit 😉

Author: I like these suggestions! Feel free to polishe me updated formulations a second time.
But if we state the intro to contour maps as highlight, I feel we should make sure that the similar "normal" tutorial for contour maps is correctly displayed (see PR#2260).

> * Created functions to download GMT remote datasets ([#2200](https://github.com/GenericMappingTools/pygmt/pull/2200))

Reviewer: Hmm, it looks like #2200 is an internal function, not sure if we want to show this to the public. I thought there was an issue tracking all of the new `load_earth_*` dataset functions done by @willschlitzer but it doesn't look like it...

I'll modify the comment in #1786 to include all the newly added functions, and then we can link to that instead.

```suggestion
* Created functions to download GMT remote datasets ([#1786](https://github.com/GenericMappingTools/pygmt/issues/1786))
```

Reviewer 4: I think this is a good change.

* Add load_earth_geoid function for Earth Geoid dataset ([#2236](https://github.com/GenericMappingTools/pygmt/pull/2236))
* Add load_earth_magnetic_anomaly function for Earth magnetic anomaly dataset ([#2196](https://github.com/GenericMappingTools/pygmt/pull/2196), [#2239](https://github.com/GenericMappingTools/pygmt/pull/2239))
* load_earth_relief: Add the support of data sources 'gebco' , 'gebcosi', and 'synbath' ([#1818](https://github.com/GenericMappingTools/pygmt/pull/1818), [#2162](https://github.com/GenericMappingTools/pygmt/pull/2162), [#2192](https://github.com/GenericMappingTools/pygmt/pull/2192))
* Wrap ternary ([#1431](https://github.com/GenericMappingTools/pygmt/pull/1431))

### Enhancements

* Add ternary sample dataset ([#2211](https://github.com/GenericMappingTools/pygmt/pull/2211))
* pygmt.config: Support tab auto-completion for all GMT defaults ([#2213](https://github.com/GenericMappingTools/pygmt/pull/2213))
* Figure.savefig: Raise a FileNotFoundError if the parent directory doesn't exist ([#2160](https://github.com/GenericMappingTools/pygmt/pull/2160))

> * Figure.psconvert: Check if the given prefix is valid ([#2170](https://github.com/GenericMappingTools/pygmt/pull/2170))
> * Figure.show: Allow keyword arguments passed to Figure.psconvert ([#2078](https://github.com/GenericMappingTools/pygmt/pull/2078))

Author: Maybe we should move PR 2319 from the "Documentations" section to the "Enhancements" section?
```suggestion
* Figure.psconvert: Check if the given prefix is valid ([#2170](https://github.com/GenericMappingTools/pygmt/pull/2170))
* Figure.ternary: Add parameters alabel, blabel, and clabel ([#2139](https://github.com/GenericMappingTools/pygmt/pull/2139))
* Figure.show: Allow keyword arguments passed to Figure.psconvert ([#2078](https://github.com/GenericMappingTools/pygmt/pull/2078))
```

> * Figure.psconvert: Add a new alias gs_path (-G) ([#2076](https://github.com/GenericMappingTools/pygmt/pull/2076))
> * Figure.ternary: Add parameters alabel, blabel, and clabel ([#2139](https://github.com/GenericMappingTools/pygmt/pull/2139))
> * Figure.psconvert: Check if the given prefix is valid ([#2170](https://github.com/GenericMappingTools/pygmt/pull/2170))

Reviewer 2: Put the two "Figure.psconvert" entries together:
```suggestion
* Figure.ternary: Add parameters alabel, blabel, and clabel ([#2139](https://github.com/GenericMappingTools/pygmt/pull/2139))
* Figure.psconvert: Add a new alias gs_path (-G) ([#2076](https://github.com/GenericMappingTools/pygmt/pull/2076))
* Figure.psconvert: Check if the given prefix is valid ([#2170](https://github.com/GenericMappingTools/pygmt/pull/2170))
```

Author: For me this change is fine. (My original idea was to sort the entries regarding added aliases alphabetically.)

* Add Yvonne to team page ([#2270](https://github.com/GenericMappingTools/pygmt/pull/2270))
* Add intro tutorial section for creating contour map ([#2126](https://github.com/GenericMappingTools/pygmt/pull/2126))
* Add inline example for grd2cpt ([#2145](https://github.com/GenericMappingTools/pygmt/pull/2145))
* SVG format is not supported in psconvert with ghostscript 9.16+ ([#2159](https://github.com/GenericMappingTools/pygmt/pull/2159))
* Add instructions to install pygmt kernel for Jupyter users ([#2153](https://github.com/GenericMappingTools/pygmt/pull/2153))
* Add inline example for grdcontour ([#2148](https://github.com/GenericMappingTools/pygmt/pull/2148))
* Add inline example for grdimage ([#2146](https://github.com/GenericMappingTools/pygmt/pull/2146))
* Add inline example for solar ([#2147](https://github.com/GenericMappingTools/pygmt/pull/2147))
* Improve instructions about setting GMT_LIBRARY_PATH env variable ([#2136](https://github.com/GenericMappingTools/pygmt/pull/2136))
* Add inline example for coast ([#2142](https://github.com/GenericMappingTools/pygmt/pull/2142))

> * Figure.ternary: Add parameters alabel, blabel, and clabel ([#2139](https://github.com/GenericMappingTools/pygmt/pull/2139))

Author: Maybe we can remove PR 2319 from the "Documentation" section and put it into to the "Enhancements" section?
```suggestion
```

* Add inline example for grd2cpt ([#2145](https://github.com/GenericMappingTools/pygmt/pull/2145))
* Add inline example for solar ([#2147](https://github.com/GenericMappingTools/pygmt/pull/2147))
* Add SciPy 2022 talk to presentations ([#2053](https://github.com/GenericMappingTools/pygmt/pull/2053))
* Add instructions to install pygmt kernel for Jupyter users ([#2153](https://github.com/GenericMappingTools/pygmt/pull/2153))
* Improve instructions about setting GMT_LIBRARY_PATH env variable ([#2136](https://github.com/GenericMappingTools/pygmt/pull/2136))
* Add badges for conda package version, license, and twitter ([#2081](https://github.com/GenericMappingTools/pygmt/pull/2081))
* Add PyOpenSci peer reviewed badge to main README.rst ([#2112](https://github.com/GenericMappingTools/pygmt/pull/2112))

### Maintenance

* Add an internal function to load GMT remote datasets ([#2200](https://github.com/GenericMappingTools/pygmt/pull/2200))

> * load_earth_relief() now returns gridline-registrated grids by default in GMT 6.4.0 ([#2052](https://github.com/GenericMappingTools/pygmt/pull/2052))

Reviewer 2: We can remove this entry because we have mentioned "Set gridline (if available) as the default grid registration for remote datasets" in the "Enhancements" section:
```suggestion
```

* Easy access to various remote datasets

### New Features

* Add load_earth_vertical_gravity_gradient function for Earth vertical gravity gradient dataset ([#2240](https://github.com/GenericMappingTools/pygmt/pull/2240))
* Add load_earth_free_air_anomaly function for Earth free-air anomaly dataset ([#2238](https://github.com/GenericMappingTools/pygmt/pull/2238))
* Add load_earth_geoid function for Earth Geoid dataset ([#2236](https://github.com/GenericMappingTools/pygmt/pull/2236))
* Add load_earth_magnetic_anomaly function for Earth magnetic anomaly dataset ([#2196](https://github.com/GenericMappingTools/pygmt/pull/2196), [#2239](https://github.com/GenericMappingTools/pygmt/pull/2239))
* load_earth_relief: Add the support of data sources 'gebco' , 'gebcosi', and 'synbath' ([#1818](https://github.com/GenericMappingTools/pygmt/pull/1818), [#2162](https://github.com/GenericMappingTools/pygmt/pull/2162), [#2192](https://github.com/GenericMappingTools/pygmt/pull/2192))
* Wrap ternary ([#1431](https://github.com/GenericMappingTools/pygmt/pull/1431))

### Enhancements

* Add ternary sample dataset ([#2211](https://github.com/GenericMappingTools/pygmt/pull/2211))

> * pygmt.config: Support tab auto-completion for all GMT defaults ([#2213](https://github.com/GenericMappingTools/pygmt/pull/2213))

Reviewer 2: What about highlighting the auto-completion of GMT defaults?

Author: I have added it to the "Highlights" section in commit https://github.com/GenericMappingTools/pygmt/pull/2272/commits/3acad82423f46e6dd85a727553e4c44df938eaf0.
I personally feel we should not have more than three or four entries in the Highlights section addional to "Eighth minor release of PyGMT".

-------------
Does point 6 of the section [Updating the Changelog](https://www.pygmt.org/dev/maintenance.html#updating-the-changelog) in the Maintainers Guide mean, that we remove the Highlight PRs from the sections below?
> Move a few important items from the main sections to the highlights section.

* Figure.text: Deprecate parameter "incols" to "use_word" (remove in v0.10.0)  ([#1964](https://github.com/GenericMappingTools/pygmt/pull/1964))

### Bug Fixes

* Figure.ternary: Fix the crash for pd.DataFrame input with GMT 6.3-6.4 ([#2274](https://github.com/GenericMappingTools/pygmt/pull/2274))
* Figure.meca: Fix line and circle of offset parameter for dict/pandas input  ([#2226](https://github.com/GenericMappingTools/pygmt/pull/2226))
* Figure.meca: Fix beachball offsetting with dict/pandas inputs ([#2202](https://github.com/GenericMappingTools/pygmt/pull/2202))
* Figure.meca: Fix the bug when passing a dict of scalar values to the spec parameter ([#2174](https://github.com/GenericMappingTools/pygmt/pull/2174))

### Documentation
* Add gallery example for Figure.ternary method ([#2138](https://github.com/GenericMappingTools/pygmt/pull/2138))
* Add Yvonne to team page ([#2270](https://github.com/GenericMappingTools/pygmt/pull/2270))
* Add intro tutorial section for creating contour map ([#2126](https://github.com/GenericMappingTools/pygmt/pull/2126))
* Add inline example for grd2cpt ([#2145](https://github.com/GenericMappingTools/pygmt/pull/2145))

> * SVG format is not supported in psconvert with ghostscript 9.16+ ([#2159](https://github.com/GenericMappingTools/pygmt/pull/2159))

Reviewer 2: ```suggestion
```

* Convert bug report markdown issue template into yaml configured form ([#2091](https://github.com/GenericMappingTools/pygmt/pull/2091))
* Convert feature request issue template into yaml configured form ([#2214](https://github.com/GenericMappingTools/pygmt/pull/2214))
* Convert module request issue template into yaml configured form ([#2216](https://github.com/GenericMappingTools/pygmt/pull/2216))
* Add support for Python 3.11 ([#2172](https://github.com/GenericMappingTools/pygmt/pull/2172))
* NEP29: Test PyGMT on NumPy 1.24 ([#2256](https://github.com/GenericMappingTools/pygmt/pull/2256))
* NEP29: Test PyGMT on NumPy 1.23 and 1.21 ([#2057](https://github.com/GenericMappingTools/pygmt/pull/2057))
* Bump the GMT version in CI to 6.4.0 ([#1990](https://github.com/GenericMappingTools/pygmt/pull/1990))
* Update baseline images for GMT 6.4 ([#1883](https://github.com/GenericMappingTools/pygmt/pull/1883))
* Migrate Continuous Documentation from Vercel to Readthedocs ([#1859](https://github.com/GenericMappingTools/pygmt/pull/1859))
* Set nested_sections to False for Sphinx-Gallery 0.11.0 regarding a correct navgation bar ([#2046](https://github.com/GenericMappingTools/pygmt/pull/2046))

> * Use the Ubuntu 22.04 docker image for readthedocs ([#2086](https://github.com/GenericMappingTools/pygmt/pull/2086))

Reviewer 2: I think this can be removed.
```suggestion
```

* pygmt.show_versions: Show GMT binary version and hide the Python interpreter path ([#1838](https://github.com/GenericMappingTools/pygmt/pull/1838))
* Add a workflow and Makefile target to test old GMT versions every Tuesday ([#2079](https://github.com/GenericMappingTools/pygmt/pull/2079))
* Move blackdoc options to pyproject.toml ([#2093](https://github.com/GenericMappingTools/pygmt/pull/2093))
* Move docformatter options from Makefile to pyproject.toml ([#2072](https://github.com/GenericMappingTools/pygmt/pull/2072))
* Add optional dependencies to pyproject.toml ([#2069](https://github.com/GenericMappingTools/pygmt/pull/2069))
* Migrate project metadata from setup.py to pyproject.toml following PEP621 ([#1848](https://github.com/GenericMappingTools/pygmt/pull/1848))
* Replace flake8 with flakeheaven ([#1847](https://github.com/GenericMappingTools/pygmt/pull/1847))
* Refactor grdview and grdimage to use virtualfile_from_data ([#1988](https://github.com/GenericMappingTools/pygmt/pull/1988))
* Use the org-wide code of conduct ([#2020](https://github.com/GenericMappingTools/pygmt/pull/2020))

### Contributors

> * [seisman](https://github.com/seisman)
> * [yvonnefroehlich](https://github.com/yvonnefroehlich)
> * [willschlitzer](https://github.com/willschlitzer)
> * [michaelgrund](https://github.com/michaelgrund)
> * [weiji14](https://github.com/weiji14)
> * [maxrjones](https://github.com/maxrjones)

Reviewer 2: ```suggestion
* [Dongdong Tian](https://github.com/seisman)
* [Yvonne Fröhlich](https://github.com/yvonnefroehlich)
* [Will Schlitzer](https://github.com/willschlitzer)
* [Michael Grund](https://github.com/michaelgrund)
* [Wei Ji Leong](https://github.com/weiji14)
* [Max Jones](https://github.com/maxrjones)
```

Author: Thanks!!

# Changelog

## Release v0.8.0 (2022/12/28)

[![Digital Object Identifier for PyGMT v0.8.0](https://zenodo.org/badge/DOI/10.5281/zenodo.7481934.svg)](https://doi.org/10.5281/zenodo.7481934)

### Highlights

* 🎉 **Eighth minor release of PyGMT** 🎉

> * Wrapped ternary
> * Easy access to various remote datasets
> * Add a intro tutorial for creating contour maps
> * Support tab auto-completion for all GMT defaults

Reviewer 4: ```suggestion
* Added support for tab auto-completion for all GMT default parameters
* Created functions to download GMT remote datasets
* Wrapped the ternary module
* Added an intro tutorial for creating contour maps
```

My take on how this should be ordered, since I think tab autocomplete is the biggest change, followed by the remote dataset functions. To keep it standardized, I think there should also be a verb of the same tense saying what happened (e.g. added, created, wrapped, etc.).

> * Wrapped ternary
> * Easy access to various remote datasets
> * Add a intro tutorial for creating contour maps
> * Support tab auto-completion for all GMT defaults

Reviewer 4: ```suggestion
* Added support for tab auto-completion for all GMT default parameters
* Created functions to download GMT remote datasets
* Wrapped the ternary module
* Added an intro tutorial for creating contour maps
```

My take on how this should be ordered, since I think tab autocomplete is the biggest change, followed by the remote dataset functions. To keep it standardized, I think there should also be a verb of the same tense saying what happened (e.g. added, created, wrapped, etc.).

Author: Totally agree @willschlitzer these formulations definitely need some polishing 🙂. Did not work on this, as I was not sure, whether everbody is happy with these highlights.

Reviewer: Would also be good to link to the PRs where these features were implemented, or the root issue in the case of the remote datasets (since there are many)!

Author: Thanks @weiji14. The links are added in commit https://github.com/GenericMappingTools/pygmt/pull/2272/commits/4573f3956745ff6f90e98957daa41c93de13839b.

* Add inline example for coast ([#2142](https://github.com/GenericMappingTools/pygmt/pull/2142))
* Add inline example for grdcontour ([#2148](https://github.com/GenericMappingTools/pygmt/pull/2148))
* Add inline example for grdimage ([#2146](https://github.com/GenericMappingTools/pygmt/pull/2146))
* Add inline example for grd2cpt ([#2145](https://github.com/GenericMappingTools/pygmt/pull/2145))
* Add inline example for solar ([#2147](https://github.com/GenericMappingTools/pygmt/pull/2147))
* Add SciPy 2022 talk to presentations ([#2053](https://github.com/GenericMappingTools/pygmt/pull/2053))
* Add instructions to install pygmt kernel for Jupyter users ([#2153](https://github.com/GenericMappingTools/pygmt/pull/2153))
* Improve instructions about setting GMT_LIBRARY_PATH env variable ([#2136](https://github.com/GenericMappingTools/pygmt/pull/2136))
* Add badges for conda package version, license and twitter ([#2081](https://github.com/GenericMappingTools/pygmt/pull/2081))
* Add PyOpenSci peer reviewed badge to main README.rst ([#2112](https://github.com/GenericMappingTools/pygmt/pull/2112))

> * Add Yvonne to team page ([#2270](https://github.com/GenericMappingTools/pygmt/pull/2270))

Reviewer 4: ```suggestion
```
Glad you are on the team :partying_face: but I don't think this needs to be included in the changelog.

# Changelog

## Release v0.8.0 (2022/12/28)

[![Digital Object Identifier for PyGMT v0.8.0](https://zenodo.org/badge/DOI/10.5281/zenodo.7481934.svg)](https://doi.org/10.5281/zenodo.7481934)

### Highlights

* 🎉 **Eighth minor release of PyGMT** 🎉
* Wrapped ternary
*

### New Features

* Add load_earth_vertical_gravity_gradient function for Earth vertical gravity gradient dataset ([#2240](https://github.com/GenericMappingTools/pygmt/pull/2240))
* Add load_earth_free_air_anomaly function for Earth free-air anomaly dataset ([#2238](https://github.com/GenericMappingTools/pygmt/pull/2238))
* Add load_earth_geoid function for Earth Geoid dataset ([#2236](https://github.com/GenericMappingTools/pygmt/pull/2236))

> * Add load_earth_magnetic_anomaly function for Earth magnetic anomaly dataset ([#2196](https://github.com/GenericMappingTools/pygmt/pull/2196))

Reviewer 2: ```suggestion
* Add load_earth_magnetic_anomaly function for Earth magnetic anomaly dataset ([#2196](https://github.com/GenericMappingTools/pygmt/pull/2196), [#2239](https://github.com/GenericMappingTools/pygmt/pull/2239))
```

* load_earth_relief: Add the support of data sources 'gebco' , 'gebcosi', and 'synbath' ([#1818](https://github.com/GenericMappingTools/pygmt/pull/1818), [#2162](https://github.com/GenericMappingTools/pygmt/pull/2162), [#2192](https://github.com/GenericMappingTools/pygmt/pull/2192))
* Wrap ternary ([#1431](https://github.com/GenericMappingTools/pygmt/pull/1431))

### Enhancements

> * load_earth_magnetic_anomaly: Add mag4km parameter to support earth_mag4km dataset ([#2239](https://github.com/GenericMappingTools/pygmt/pull/2239))

Reviewer 2: I think this entry can be deleted because the `load_earth_magnetic_anomaly` function is also newly added in this release.
```suggestion
```

* Add ternary sample dataset ([#2211](https://github.com/GenericMappingTools/pygmt/pull/2211))
* pygmt.config: Support tab auto-completion for all GMT defaults ([#2213](https://github.com/GenericMappingTools/pygmt/pull/2213))

> * Figure.text: Deprecate parameter "incols" to "use_word" (remove in v0.10.0)  ([#1964](https://github.com/GenericMappingTools/pygmt/pull/1964))

Reviewer 2: This should be moved to the "Deprecataions" section:
```suggestion
```

* Figure.savefig: Raise a FileNotFoundError if the parent directory doesn't exist ([#2160](https://github.com/GenericMappingTools/pygmt/pull/2160))
* Figure.psconvert: Check if the given prefix is valid ([#2170](https://github.com/GenericMappingTools/pygmt/pull/2170))
* Figure.show: Allow keyword arguments passed to Figure.psconvert ([#2078](https://github.com/GenericMappingTools/pygmt/pull/2078))
* Figure.psconvert: Add a new alias gs_path (-G) ([#2076](https://github.com/GenericMappingTools/pygmt/pull/2076))
* Rewrite the meca function to support offsetting and labeling beachballs ([#1784](https://github.com/GenericMappingTools/pygmt/pull/1784))

### Deprecations

> 

Reviewer 2: ```suggestion
* Figure.wiggle: Deprecate parameter "color" (remove in v0.12.0) and add "fillpositive"/"fillnegative" ([#2205](https://github.com/GenericMappingTools/pygmt/pull/2205))
```

* Figure.psconvert: Remove the deprecated parameter 'icc_gray' (deprecated since v0.6.0) ([#2267](https://github.com/GenericMappingTools/pygmt/pull/2267))
* Figure.velo: Deprecate parameters "color" to "fill" and "uncertaintycolor" to "uncertaintyfill" (remove in v0.12.0) ([#2206](https://github.com/GenericMappingTools/pygmt/pull/2206))
* Deprecate xshift (X) and yshift (Y) aliases from all plotting modules (remove in v0.12.0) ([#2071](https://github.com/GenericMappingTools/pygmt/pull/2071))
* Figure.rose: Deprecate parameter color to fill (remove in v0.12.0) ([#2181](https://github.com/GenericMappingTools/pygmt/pull/2181))
* Figure.plot3d: Deprecate parameter "color" to "fill" (remove in v0.12.0) ([#2178](https://github.com/GenericMappingTools/pygmt/pull/2178))

> * Figure.plot: Deprecate parameter "color" to "fill" (remove in v0.12.0) ([#2177](https://github.com/GenericMappingTools/pygmt/pull/2177))

Reviewer 2: ```suggestion
* Figure.plot: Deprecate parameter "color" to "fill" (remove in v0.12.0) ([#2177](https://github.com/GenericMappingTools/pygmt/pull/2177))
* Figure.text: Deprecate parameter "incols" to "use_word" (remove in v0.10.0)  ([#1964](https://github.com/GenericMappingTools/pygmt/pull/1964))
```

### Bug Fixes

* Figure.ternary: Fix the crash for pd.DataFrame input with GMT 6.3-6.4 ([#2274](https://github.com/GenericMappingTools/pygmt/pull/2274))
* Figure.meca: Fix line and circle of offset parameter for dict/pandas input  ([#2226](https://github.com/GenericMappingTools/pygmt/pull/2226))
* Figure.meca: Fix beachball offsetting with dict/pandas inputs ([#2202](https://github.com/GenericMappingTools/pygmt/pull/2202))
* Figure.meca: Fix the bug when passing a dict of scalar values to the spec parameter ([#2174](https://github.com/GenericMappingTools/pygmt/pull/2174))

### Documentation

> 

Reviewer 2: ```suggestion
* Add gallery example for Figure.ternary method ([#2138](https://github.com/GenericMappingTools/pygmt/pull/2138))
```

* Add instructions to install pygmt kernel for Jupyter users ([#2153](https://github.com/GenericMappingTools/pygmt/pull/2153))
* Add inline example for grdcontour ([#2148](https://github.com/GenericMappingTools/pygmt/pull/2148))
* Add inline example for grdimage ([#2146](https://github.com/GenericMappingTools/pygmt/pull/2146))
* Add inline example for solar ([#2147](https://github.com/GenericMappingTools/pygmt/pull/2147))
* Improve instructions about setting GMT_LIBRARY_PATH env variable ([#2136](https://github.com/GenericMappingTools/pygmt/pull/2136))
* Add inline example for coast ([#2142](https://github.com/GenericMappingTools/pygmt/pull/2142))
* Figure.ternary: Add parameters alabel, blabel, and clabel ([#2139](https://github.com/GenericMappingTools/pygmt/pull/2139))
* Convert bug report markdown issue template into yaml configured form ([#2091](https://github.com/GenericMappingTools/pygmt/pull/2091))
* Add a gallery example showing the usage of vertical and horizontal bars ([#1521](https://github.com/GenericMappingTools/pygmt/pull/1521))
* Add PyOpenSci peer reviewed badge to main README.rst ([#2112](https://github.com/GenericMappingTools/pygmt/pull/2112))

> * Correct and consistent usage of 'function, 'method', and 'class' ([#2043](https://github.com/GenericMappingTools/pygmt/pull/2043))

Reviewer 2: ```suggestion
```

* doc: Set different html_baseurl for stable and dev versions ([#2158](https://github.com/GenericMappingTools/pygmt/pull/2158))
* Use longname placeholders in the docstrings for common options ([#1932](https://github.com/GenericMappingTools/pygmt/pull/1932))
* pygmt.show_versions: Show GMT binary version and hide the Python interpreter path ([#1838](https://github.com/GenericMappingTools/pygmt/pull/1838))
* Add a workflow and Makefile target to test old GMT versions every Tuesday ([#2079](https://github.com/GenericMappingTools/pygmt/pull/2079))
* Move blackdoc options to pyproject.toml ([#2093](https://github.com/GenericMappingTools/pygmt/pull/2093))
* Move docformatter options from Makefile to pyproject.toml ([#2072](https://github.com/GenericMappingTools/pygmt/pull/2072))
* Use the Ubuntu 22.04 docker image for readthedocs ([#2086](https://github.com/GenericMappingTools/pygmt/pull/2086))
* Add badges for conda package version, license and twitter ([#2081](https://github.com/GenericMappingTools/pygmt/pull/2081))
* Add optional dependencies to pyproject.toml ([#2069](https://github.com/GenericMappingTools/pygmt/pull/2069))

> * Remove tomli from GMT Tests and GMT Dev Tests workflows ([#2080](https://github.com/GenericMappingTools/pygmt/pull/2080))

Reviewer 2: ```suggestion
```

* Replace flake8 with flakeheaven ([#1847](https://github.com/GenericMappingTools/pygmt/pull/1847))
* NEP29: Test PyGMT on NumPy 1.23 and 1.21 ([#2057](https://github.com/GenericMappingTools/pygmt/pull/2057))
* load_earth_relief() now returns gridline-registrated grids by default in GMT 6.4 ([#2052](https://github.com/GenericMappingTools/pygmt/pull/2052))
* Set nested_sections to False for Sphinx-Gallery 0.11.0 regarding a correct navgation bar ([#2046](https://github.com/GenericMappingTools/pygmt/pull/2046))
* Update baseline images for GMT 6.4 ([#1883](https://github.com/GenericMappingTools/pygmt/pull/1883))
* Bump the GMT version in CI to 6.4.0 ([#1990](https://github.com/GenericMappingTools/pygmt/pull/1990))
* Migrate Continuous Documentation from Vercel to Readthedocs ([#1859](https://github.com/GenericMappingTools/pygmt/pull/1859))
* Use the org-wide code of conduct ([#2020](https://github.com/GenericMappingTools/pygmt/pull/2020))
* Refactor grdview and grdimage to use virtualfile_from_data ([#1988](https://github.com/GenericMappingTools/pygmt/pull/1988))

> * Use consistent names for pytest fixtures ([#1965](https://github.com/GenericMappingTools/pygmt/pull/1965))

Reviewer 2: ```suggestion
```

# Changelog

## Release v0.8.0 (2022/12/28)

[![Digital Object Identifier for PyGMT v0.8.0](https://zenodo.org/badge/DOI/10.5281/zenodo.7481934.svg)](https://doi.org/10.5281/zenodo.7481934)

### Highlights

* 🎉 **Eighth minor release of PyGMT** 🎉
* Wrapped ternary
*

### New Features

* Add load_earth_vertical_gravity_gradient function for Earth vertical gravity gradient dataset ([#2240](https://github.com/GenericMappingTools/pygmt/pull/2240))
* Add load_earth_free_air_anomaly function for Earth free-air anomaly dataset ([#2238](https://github.com/GenericMappingTools/pygmt/pull/2238))
* Add load_earth_geoid function for Earth Geoid dataset ([#2236](https://github.com/GenericMappingTools/pygmt/pull/2236))
* Add load_earth_magnetic_anomaly function for Earth magnetic anomaly dataset ([#2196](https://github.com/GenericMappingTools/pygmt/pull/2196))

> * load_earth_relief: Add the support of data source 'GEBCOSI' ([#2192](https://github.com/GenericMappingTools/pygmt/pull/2192))
> * load_earth_relief: Add the support of data source 'synbath' ([#2162](https://github.com/GenericMappingTools/pygmt/pull/2162))
> * load_earth_relief: Add the support of data source "GEBCO" ([#1818](https://github.com/GenericMappingTools/pygmt/pull/1818))

Reviewer 2: ```suggestion
* load_earth_relief: Add the support of data sources 'gebco' ([#1818](https://github.com/GenericMappingTools/pygmt/pull/1818)), 'gebcosi' ([#2192](https://github.com/GenericMappingTools/pygmt/pull/2192)), and 'synbath' ([#2162](https://github.com/GenericMappingTools/pygmt/pull/2162))
```

* Add intro tutorial section for creating contour map ([#2126](https://github.com/GenericMappingTools/pygmt/pull/2126))
* Add inline example for grd2cpt ([#2145](https://github.com/GenericMappingTools/pygmt/pull/2145))
* SVG format is not supported in psconvert with ghostscript 9.16+ ([#2159](https://github.com/GenericMappingTools/pygmt/pull/2159))
* Add instructions to install pygmt kernel for Jupyter users ([#2153](https://github.com/GenericMappingTools/pygmt/pull/2153))
* Add inline example for grdcontour ([#2148](https://github.com/GenericMappingTools/pygmt/pull/2148))
* Add inline example for grdimage ([#2146](https://github.com/GenericMappingTools/pygmt/pull/2146))
* Add inline example for solar ([#2147](https://github.com/GenericMappingTools/pygmt/pull/2147))
* Improve instructions about setting GMT_LIBRARY_PATH env variable ([#2136](https://github.com/GenericMappingTools/pygmt/pull/2136))
* Add inline example for coast ([#2142](https://github.com/GenericMappingTools/pygmt/pull/2142))
* Figure.ternary: Add parameters alabel, blabel, and clabel ([#2139](https://github.com/GenericMappingTools/pygmt/pull/2139))

> * Improve tutorial frames.py ([#2023](https://github.com/GenericMappingTools/pygmt/pull/2023))

Reviewer 2: ```suggestion
```

* Convert bug report markdown issue template into yaml configured form ([#2091](https://github.com/GenericMappingTools/pygmt/pull/2091))
* Add a gallery example showing the usage of vertical and horizontal bars ([#1521](https://github.com/GenericMappingTools/pygmt/pull/1521))
* Add PyOpenSci peer reviewed badge to main README.rst ([#2112](https://github.com/GenericMappingTools/pygmt/pull/2112))
* Correct and consistent usage of 'function, 'method', and 'class' ([#2043](https://github.com/GenericMappingTools/pygmt/pull/2043))
* Add SciPy 2022 talk to presentations ([#2053](https://github.com/GenericMappingTools/pygmt/pull/2053))

> * Improve documentation of UTM projection ([#2061](https://github.com/GenericMappingTools/pygmt/pull/2061))
> * Consistent usage of whitespaces across gallery and tutorial examples ([#2018](https://github.com/GenericMappingTools/pygmt/pull/2018))

Reviewer 2: ```suggestion
```

### Maintenance

* Add support for Python 3.11 ([#2172](https://github.com/GenericMappingTools/pygmt/pull/2172))

> * Do not show the 'Improve this page' button on sphinx-gallery execution time pages ([#2264](https://github.com/GenericMappingTools/pygmt/pull/2264))

Reviewer 2: I think this entry can be deleted
```suggestion
```
